<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ioInterface</class>
 <widget class="QTabWidget" name="ioInterface">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>8</pointsize>
   </font>
  </property>
  <property name="focusPolicy">
   <enum>Qt::TabFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Rpi io interface</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="currentIndex">
   <number>2</number>
  </property>
  <property name="iconSize">
   <size>
    <width>40</width>
    <height>16</height>
   </size>
  </property>
  <widget class="QWidget" name="Outputs">
   <attribute name="title">
    <string>Conveyor</string>
   </attribute>
   <widget class="QLabel" name="Output_Title">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>10</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Conveyor control</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_15">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>50</y>
      <width>31</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="toolTip">
     <string>Card stuck sensor</string>
    </property>
    <property name="toolTipDuration">
     <number>0</number>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 0, 0);
background-color: rgb(255, 255, 127);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="btnFwd">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>110</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>FWD</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnStop">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>110</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>STOP</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnRev">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>REV</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnReset">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>170</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_idxcounter">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>50</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="btnCONVEYOR_THREAD">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>110</y>
      <width>111</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>Conveyor thread</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnESTOP">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>110</y>
      <width>111</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>Estop</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="btnEnable">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>230</y>
      <width>121</width>
      <height>91</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>15</width>
      <height>15</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox::indicator {
width: 20px;
height: 20px;
}
</string>
    </property>
    <property name="text">
     <string>Enable</string>
    </property>
    <property name="iconSize">
     <size>
      <width>28</width>
      <height>28</height>
     </size>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="btnEnableDispense">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>230</y>
      <width>121</width>
      <height>91</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>15</width>
      <height>15</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox::indicator {
width: 20px;
height: 20px;
}
</string>
    </property>
    <property name="text">
     <string>Dispense</string>
    </property>
    <property name="iconSize">
     <size>
      <width>28</width>
      <height>28</height>
     </size>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="btnEnableEject">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>230</y>
      <width>121</width>
      <height>91</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>15</width>
      <height>15</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox::indicator {
width: 20px;
height: 20px;
}
</string>
    </property>
    <property name="text">
     <string>Eject</string>
    </property>
    <property name="iconSize">
     <size>
      <width>28</width>
      <height>28</height>
     </size>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="lblConveyor_FSM">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>220</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>State</string>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="tab">
   <attribute name="title">
    <string>Dispenser</string>
   </attribute>
   <widget class="QPushButton" name="btnDispense">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>111</width>
      <height>111</height>
     </rect>
    </property>
    <property name="text">
     <string>Dispense</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnGetStatus">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>40</y>
      <width>111</width>
      <height>111</height>
     </rect>
    </property>
    <property name="text">
     <string>Get status</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblDispenserStatus">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>40</y>
      <width>481</width>
      <height>101</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>34</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Status...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnReset_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>160</y>
      <width>111</width>
      <height>111</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnInit">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>160</y>
      <width>111</width>
      <height>111</height>
     </rect>
    </property>
    <property name="text">
     <string>Initialize</string>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="Inputs">
   <attribute name="title">
    <string>Bins</string>
   </attribute>
   <widget class="QListWidget" name="Input_Status">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>20</y>
      <width>271</width>
      <height>251</height>
     </rect>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_65">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>80</y>
      <width>31</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_66">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>290</y>
      <width>31</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_81">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>20</y>
      <width>31</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="toolTip">
     <string>Card stuck sensor</string>
    </property>
    <property name="toolTipDuration">
     <number>0</number>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 0, 0);
background-color: rgb(255, 255, 127);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="btnUP">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>80</y>
      <width>50</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>UP</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="autoRepeat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="btnDN">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>280</y>
      <width>50</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>DN</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_0">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>300</y>
      <width>31</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="btnAUTO_THREAD">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>AUTO Cycle bin1</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QSlider" name="servoStart">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>50</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimum">
     <number>150</number>
    </property>
    <property name="maximum">
     <number>600</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
   </widget>
   <widget class="QSlider" name="servoStop">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>90</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimum">
     <number>150</number>
    </property>
    <property name="maximum">
     <number>600</number>
    </property>
    <property name="value">
     <number>600</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="servoStartAngle">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>30</y>
      <width>64</width>
      <height>23</height>
     </rect>
    </property>
    <property name="digitCount">
     <number>3</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>150.000000000000000</double>
    </property>
   </widget>
   <widget class="QLCDNumber" name="servoStopAngle">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>130</y>
      <width>64</width>
      <height>23</height>
     </rect>
    </property>
    <property name="digitCount">
     <number>3</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>600.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="btnEject">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>20</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>eject</string>
    </property>
   </widget>
   <widget class="QLabel" name="Input_Title_2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>30</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Servo control</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="CycleCountBin1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QSlider" name="bin1Level">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>80</y>
      <width>41</width>
      <height>251</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 85, 0);</string>
    </property>
    <property name="maximum">
     <number>6000</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
    <property name="tickInterval">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>60</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>High Limit</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>340</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Low Limit</string>
    </property>
   </widget>
   <widget class="QDial" name="bin1Level_set">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>220</y>
      <width>111</width>
      <height>111</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-700</number>
    </property>
    <property name="maximum">
     <number>700</number>
    </property>
   </widget>
   <widget class="QLCDNumber" name="bin1Level_target">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>170</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="btnMANUAL_THREAD">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>230</y>
      <width>111</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>MANUAL bin1</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>80</y>
      <width>16</width>
      <height>251</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="minimum">
     <number>70</number>
    </property>
    <property name="maximum">
     <number>470</number>
    </property>
    <property name="value">
     <number>69</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
    <property name="invertedAppearance">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_82">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>20</y>
      <width>31</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="toolTip">
     <string>Card stuck sensor</string>
    </property>
    <property name="toolTipDuration">
     <number>0</number>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 0, 0);
background-color: rgb(255, 255, 127);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="midLineWidth">
     <number>0</number>
    </property>
    <property name="digitCount">
     <number>1</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Flat</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="lblBin1_FSM">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>State</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="tofDistance">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>340</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QCheckBox" name="btnEnableCycle">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>140</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="baseSize">
     <size>
      <width>15</width>
      <height>15</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox::indicator {
width: 20px;
height: 20px;
}
</string>
    </property>
    <property name="text">
     <string>Enabled</string>
    </property>
    <property name="iconSize">
     <size>
      <width>28</width>
      <height>28</height>
     </size>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QScrollBar" name="topLevel">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>80</y>
      <width>16</width>
      <height>251</height>
     </rect>
    </property>
    <property name="maximum">
     <number>600</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="tofDistanceSP">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>340</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>servoStart</sender>
   <signal>valueChanged(int)</signal>
   <receiver>servoStartAngle</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>360</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>servoStop</sender>
   <signal>valueChanged(int)</signal>
   <receiver>servoStopAngle</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>344</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnESTOP</sender>
   <signal>clicked()</signal>
   <receiver>ioInterface</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>332</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bin1Level_set</sender>
   <signal>valueChanged(int)</signal>
   <receiver>bin1Level_target</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>346</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>357</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>progressBar</sender>
   <signal>valueChanged(int)</signal>
   <receiver>tofDistance</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topLevel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>tofDistanceSP</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>273</x>
     <y>375</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
